solr/modules/jwt-auth/src/test/org/apache/solr/security/jwt/JWTAuthPluginIntegrationTest.java

```diff
@@ -116,6 +122,12 @@ public static void afterClass() throws Exception {
     if (mockOAuth2Server != null) {
       mockOAuth2Server.shutdown();
     }
+    if (tempDir != null) {
+      Files.walk(tempDir)
+              .sorted(Comparator.reverseOrder())
+              .map(Path::toFile)
+              .forEach(java.io.File::delete);
```

Comment:
I think you can replace all this with `tempDir.toFile().deleteOnExit()`?

Reply:
Have you tried? I think it will clean up even non-empty dirs

Reply:
I pushed the simplification using `deleteOnExit`. We use it lots of other places. If there is indeed some issue with it, feel free to revert.